python/ray/tune/execution/trial_runner.py

```diff
@@ -1433,37 +1442,9 @@ def _try_recover(self, trial: Trial, exc: Union[TuneError, RayTaskError]):
             error=exc is not None,
             exc=exc,
         )
-        if self.trial_executor.has_resources_for_trial(trial):
-            requeue_trial = False
-            logger.info(
-                "Trial %s: Attempting to restore trial state from last checkpoint.",
-                trial,
-            )
-            # TODO(xwjiang): For better consistency, consider not starting
-            #  trials here. Instead rely on requeuing the trial.
-            started = self.trial_executor.start_trial(trial)
-            if not started:
-                requeue_trial = True
-            elif trial.status == Trial.ERROR:
```

Comment:
This additional branch here violates the restore retry logic, better to consolidate to using just the re-queue mechanism.